### PR TITLE
Add v0.2.0 memeoooorr factory contracts, events, updates

### PR DESCRIPTION
**Pull request by truemiller**
Added latest ABIs
Added latest contract addresses for Base & Celo factories
Updated schema tables to support both memeTokenId (address) and memeTokenNonce (uint)
Added docker compose for development with postgres

### CODE REVIEW

File: src/Memeorr.ts [hunk 1]
```diff
@@ -340,3 +328,387 @@ ponder.on("MemeCelo:Unleashed", async ({ event, context }) => {
...
+      memeToken: event.args.memeToken,
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+});
+
+ponder.on("MemeBase_0_2_0:Summoned", async ({ event, context }) => {
+  try {
+    await context.db.totalHeartAmount.create({
```
Comment (truemiller) [inline src/Memeorr.ts:584]: review `id`

File: src/Memeorr.ts [hunk 2]
```diff
@@ -340,3 +328,387 @@ ponder.on("MemeCelo:Unleashed", async ({ event, context }) => {
...
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+});
+
+ponder.on("MemeCelo_0_2_0:Summoned", async ({ event, context }) => {
+  try {
+    await context.db.totalHeartAmount.create({
+      id: `${event.args.memeNonce}`,
```
Comment (truemiller) [inline src/Memeorr.ts:625]: review `id`

File: src/Memeorr.ts [hunk 3]
```diff
@@ -340,3 +328,387 @@ ponder.on("MemeCelo:Unleashed", async ({ event, context }) => {
...
+      data: {
+        chain: "celo",
+        amount: 0n,
+      },
+    });
+    await context.db.MemeToken.create({
+      id: event.args.memeNonce.toString(),
```
Comment (truemiller) [inline src/Memeorr.ts:632]: review `id`, event no longer provides a clear `id`

File: ponder.schema.ts [hunk 4]
```diff
@@ -73,13 +89,15 @@ export default createSchema((p) => ({
     id: p.string(),
     chain: p.string(),
     hearter: p.string(),
-    memeTokenId: p.string().references("MemeToken.id"),
+    memeTokenId: p.string().references("MemeToken.id").optional(),
+    memeNonce: p.bigint().optional(),
```
Comment (truemiller) [inline ponder.schema.ts:93]: unable to reference here, unsure if this foreign keys database, or if is an "API layer" reference

Comment (DavidMinarsch) [reply ponder.schema.ts:93]: This should be memeToken no? In the old contract the Heart event had that as the second value of the event, now its memeNonce

File: src/Memeorr.ts [hunk 5]
```diff
@@ -340,3 +328,387 @@ ponder.on("MemeCelo:Unleashed", async ({ event, context }) => {
...
+      chain: "base",
+      hearter: event.args.hearter,
+      memeNonce: event.args.memeNonce,
+      amount: event.args.amount,
+      timestamp: Number(event.block.timestamp),
+      blockNumber: Number(event.block.number),
+    },
+  });
+  try {
+    await context.db.Heart.create({
```
Comment (DavidMinarsch) [inline src/Memeorr.ts:407]: why repeated?

File: src/Memeorr.ts [hunk 6]
```diff
@@ -340,3 +328,387 @@ ponder.on("MemeCelo:Unleashed", async ({ event, context }) => {
...
+        chain: "base",
+        hearter: event.args.hearter,
+        memeNonce: event.args.memeNonce,
+        amount: event.args.amount,
+        timestamp: Number(event.block.timestamp),
+        blockNumber: Number(event.block.number),
+      },
+    });
+
+    // cannot specifiy memeNonce as unique, opt for findMany and limit 1
```
Comment (DavidMinarsch) [inline src/Memeorr.ts:419]: why? it's unique on contract level

File: src/Memeorr.ts (hunk 1 above)
Comment (DavidMinarsch) [reply src/Memeorr.ts:584]: summoners can summon multiple times, not sure this makes sense

File: ponder.schema.ts (hunk 4 above)
Comment (truemiller) [reply ponder.schema.ts:93]: agree, this was renamed previously, but confusing, removing in follow up

**Pull request merged**